File: Backend/src/main/java/com/persona/Backend/Entity/Parameter/Infraestrutura/Regional.java
package com.persona.Backend.Entity.Parameter.Infraestrutura;

import com.persona.Backend.Entity.BaseEntity;
import com.persona.Backend.Entity.Parameter.Ubicacion.Departamento;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "regional")
public class Regional extends BaseEntity {

	@Column(name = "nit", length = 45, nullable = false)
	private String nit;

	@Column(name = "nombre", length = 45, nullable = false)
	private String nombre;

	@Column(name = "acronimo", length = 45, nullable = false)
	private String acronimo;

	@Column(name = "direccion", length = 45, nullable = false)
	private String direccion;

	@Column(name = "telefono", length = 45, nullable = false)
	private String telefono;

	@OneToOne(fetch = FetchType.EAGER, optional = false)
	@JoinColumn(name = "departamento_id", nullable = false, unique = false)
	private Departamento departamentoId;

	public String getNit() {
		return nit;
	}

	public void setNit(String nit) {
		this.nit = nit;
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public String getAcronimo() {
		return acronimo;
	}

	public void setAcronimo(String acronimo) {
		this.acronimo = acronimo;
	}

	public String getDireccion() {
		return direccion;
	}

	public void setDireccion(String direccion) {
		this.direccion = direccion;
	}

	public String getTelefono() {
		return telefono;
	}

	public void setTelefono(String telefono) {
		this.telefono = telefono;
	}

	public Departamento getDepartamentoId() {
		return departamentoId;
	}

	public void setDepartamentoId(Departamento departamentoId) {
		this.departamentoId = departamentoId;
	}

}

File: ScheduleFrontend/src/app/models/M-Parameter/infraestructura/regional.ts
export interface Regional {
    id: number;
    state: boolean;
    createdAt?: string;
    updatedAt?: string;
    deletedAt?: string | null;
    nit: string;
    nombre: string;
    acronimo: string;
    direccion: string;
    telefono: string;
    departamentoId: {
      id: number;
      state: boolean;
      createdAt?: string;
      updatedAt?: string;
      deletedAt?: string | null;
      nombre: string;
      codigo: string;
      paisId: {
        id: number;
        state: boolean;
        createdAt?: string;
        updatedAt?: string;
        deletedAt?: string | null;
        nombre: string;
        codigo: string;
        continenteId: {
          id: number;
          state: boolean;
          createdAt?: string;
          updatedAt?: string;
          deletedAt?: string | null;
          nombre: string;
          codigo: string;
        };
      };
    };
  }
  